### ISSUE TITLE
Linting (and blowing up) on component that isn't a styled component

### ISSUE THREAD

**Issue by mcmillion** [styled-components/stylelint-processor-styled-components#246]
I'm assuming this is related to the fact that `stylelint-processor-styled-components` uses its own Babel config and we're using `transform-object-rest-spread` in our project.  I've read through the issues regarding configuring this project's Babel options, but I don't think that helps me here. Furthermore, I can't find a way to temporarily ignore this file other than adding it explicitly to `ignoreFiles`.

Try to lint this file:

```jsx
import React from 'react'
import styled from 'styled-components'

const StyledThing = styled.div`
  color: red;
`;

export const NonStyledThing = ({ whatever, ...otherArgs = {} }) => {
  return { whatever, ...otherArgs };
}
```

## Expected Behavior

Things lint

## Actual Behavior

The linting of this file blows up with `(8:46): Invalid rest operator's argument`. This seems to happen when the function component is parsed, so even if the file doesn't have any styled components it still happens.

**Comment by chinesedfan** [comment]
Hi, can your project parse the file correctly, without linting? It seems that even with `transform-object-rest-spread`, babel will throw an error. See [my test](https://astexplorer.net/#/gist/c2b5dfc8e5787c241a5deb443094153f/abcc996fb7ed657bf4bc1a13487f8e5032303012).

If removing the default value, it is OK. Both in babel and our processor. So I think it is due to syntax.

```
export const NonStyledThing = ({ whatever, ...otherArgs }) => {
  return { whatever, ...otherArgs };
}
```
